src/app/api/auth/[...nextauth]/options.ts

```diff
@@ -0,0 +1,56 @@
+import NextAuth, {
+  GhExtendedProfile,
+  NextAuthOptions,
+  Session,
+} from "next-auth";
+import GithubProvider from "next-auth/providers/github";
+
+import { githubClientID, githubClientSecret } from "@/config/process";
+import { signUp } from "@/services/core/auth";
+
+export const authOptions: NextAuthOptions = {
+  providers: [
+    GithubProvider({
+      clientId: githubClientID,
+      clientSecret: githubClientSecret,
+    }),
+  ],
+  callbacks: {
+    async jwt({ token, account, ...response }) {
+      const profile = response.profile as GhExtendedProfile | undefined;
+      if (!token?.id && account?.access_token && profile) {
+        try {
+          const { session_token, user_id } = await signUp({
+            username: profile?.login,
+            github_username: profile?.login,
+            email: profile?.email,
+            profile_pic_url: profile?.avatar_url,
+            role: "user",
+            provider: "github",
+          });
+          token.accessToken = session_token;
```

Comment:
```suggestion
          });
          token.id = user_id;
          token.accessToken = session_token;
```

We should set the token ID on a successful signup to prevent repeat signups.